### PR TITLE
Implement the I18nPlugin

### PR DESCRIPTION
**Pull request by mwistrand**
**Type:** feature

The following has been addressed in the PR:

* [ ] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**

Implements an `I18nPlugin` that creates a custom module that sets the locale and CLDR data, and ensures that module is imported into the specified entry point. This has been verified on both Windows and Mac with `todo-mvc-kitchensink` (specifically [this branch](https://github.com/mwistrand/examples/tree/i18n-updates)). While these changes cannot be incorporated into `@dojo/cli-build-webpack` until changes to both `@dojo/widget-core` and `@dojo/i18n` have been released, this PR by itself does not introduce any side effects into the build.

### CODE REVIEW

File: src/memory-source-plugin/MemorySourcePlugin.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
+import uuid from '@dojo/core/uuid';
```
Comment (matt-gadd) [inline src/memory-source-plugin/MemorySourcePlugin.ts]: is this module used for anything other than the tests?

**Comment by mwistrand** [comment]
Yes, Webpack needs a path for any module it bundles. This plugin overrides the compiler’s file system reader to return the specified source when it encounters a particular mid, and the uuid is used as that mid.

**Comment by matt-gadd** [comment]
I'm a bit confused sorry 😢. Do you think you can give me a run down of what some of this stuff on a high level does? 

When I added support in cli-next for service workers which I think has similarities in terms of whats going on here it didn't require me writing plugins to support it. 

For the service worker support:
* I included dependencies in the entry if they were needed to be injected in the build
* For a templated dependency I included that in the entry then used the `DefinePlugin` to inject whatever I needed to (in my case things from a `.dojorc`). This had the advantage of me being able to actually write typescript for that templated dependency rather than a string which had to work down to es5.

**Comment by mwistrand** [comment]
@matt-gadd After clearing up some confusion on my part, I was able to remove the `MemorySourcePlugin` and move the template string into a dedicated TS module (although I had to be careful not to use any es6 features as `webpack-contrib` does not transpile to es5). I also replaced data injection with the `DefinePlugin` (thanks for pointing that out, I always forget about that).

**Pull request merged**